server/api/auth_handler.py

```diff
@@ -0,0 +1,168 @@
+                }
+                resp = make_response(jsonify(responseObject))
+                resp.set_cookie("Authentication token", auth_token.decode(), httponly=True)  # Should the auth_token be encoded or decoded?
+                return resp, 201
+            except Exception as e:
+                responseObject = {
+                    "status": "fail",
+                    "message": "Some error occurred. Please try again."
+                }
+                return make_response(jsonify(responseObject)), 401
+        else:
+            responseObject = {
+                "status": "fail",
+                "message": "User already exists. Please log in."
+            }
+            return make_response(jsonify(responseObject)), 409
+
+
+class LoginAPI(MethodView):
+    def post(self):
+        post_data = request.get_json()
+        try:
+            user = User.query.filter_by(
+                email=post_data.get("email")
+            ).first()
+            if user and flask_bcrypt.check_password_hash(
+                user.password, post_data.get("password")
+            ):
+                auth_token = user.encode_auth_token(user.id)
+                if auth_token:
+                    g.user = user
+                    responseObject = {
+                        "status": "success",
+                        "message": "Successfully logged in",
+                        "auth_token": auth_token.decode()
+                    }
+                    resp = make_response(jsonify(responseObject))
+                    resp.set_cookie("Authentication token", auth_token.decode(), httponly=True)  # Should the auth_token be encoded or decoded?
+                    return resp, 200
+                else:
+                    responseObject = {
+                        "status": "fail",
+                        "message": "User does not exist."
+                    }
+                    return make_response(jsonify(responseObject)), 404
+            elif user:
+                responseObject = {
+                    "status": "fail",
+                    "message": "Password is incorrect."
+                }
+                return make_response(jsonify(responseObject)), 401
+            else:
+                responseObject = {
+                    "status": "fail",
+                    "message": "User does not exist."
+                }
+                return make_response(jsonify(responseObject)), 404
+        except Exception as e:
+            responseObject = {
+                "status": "fail",
+                "message": "Try again"
+            }
+            return make_response(jsonify(responseObject)), 500
+
+
+class UserAPI(MethodView):
+    def get(self):
+        auth_header = request.headers.get("Authorization")
+        if auth_header:
+            auth_token = auth_header.split(" ")[1]
+        else:
+            auth_token = ""
+        if auth_token:
+            resp = User.decode_auth_token(auth_token)
```

Comment:
Here you would actually check the request.cookies (you'd look for a cookie with the same name as the one you set) rather than the headers. But let's not do this inside each route as that will get repetitive. Instead, use a decorator that does that and can be reused across requests.